
**Python Blok Projesi**<p><img style="float: right;margin:5px 20px 5px 1px; max-width:250px" src="https://assets.datacamp.com/production/project_441/img/Nobel_Prize.png"></p>

<p>Nobel Ödülü belki de dünyanın en tanınmış bilimsel ödülüdür. Her yıl kimya, edebiyat, fizik, tıp, ekonomi ve barış alanlarında bilim insanlarına ve akademisyenlere verilmektedir. Bu projede, Nobel Ödülü kazananları inceleyeceğiz. </p>

## Proje Soruları

1. Nobel Ödüllerini en çok kazanan ilk on ülkeyi bulunuz.
2. Nobel Ödüllerini kazanan ilk kadınları listeleyiniz.
3. Nobel Ödüllerini kazanan ilk erkekleri listeleyiniz.
4. Nobel ödülünü en çok kazanan ülkenin hangi yıldan itibaren hakimiyet sağladığını görselleştirip bu hakimiyette rol oynayan şeyler nelerdir? İçgörülerinizi paylaşır mısınız?
5. Nobel Ödülü kazananların cinsiyetlerini, yaşlarını, ödül kategorisi ve yılları kullanarak görselleştiriniz.(Her bir ödül kategorisi için ayrı grafik gösteriniz) Çıkan sonuçlara göre görseli yorumlayınız.
6. 1938-1945 yılı arasında Nobel Ödülü kazananların kategorilerini ve ülkelerini görselleştirip yorumlayınız.
7. 1947-1991 yılları arasında Nobel Ödülü kazananların kategorilerini ve ülkelerini görselleştirip yorumlayınız.(Her kategori için ayrı bir grafik olması istenmektedir)
8. Kimya, Edebiyat, Barış, Fizik ve Tıp kategorilerindeki 2000 sonrasındaki kişilerin ülkelerini, yaşlarını görselleştirin.(Her bir Kategori için ayrı görselleştirme yapılması istenmektedir) Veriyi yorumlayınız.


## PROJE BAŞLANGICI:

### 1- Veri ve kütüphane yükleme ve Verinin analiz için kullanıma hazırlanması

In [46]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [47]:
# Kullanılacak kütüphaneler yüklendi

import pandas as pd
import matplotlib.pyplot as plt
import plotly.express as px

In [48]:
# Nobel csv excel dosyasını okunması.

nobel = pd.read_csv('/content/asset-v1_Kodlasam+MS01+2023_06+type@asset+block@nobel.csv')
nobel.head(10)

year    category                                           prize  \
0  1901   Chemistry               The Nobel Prize in Chemistry 1901   
1  1901  Literature              The Nobel Prize in Literature 1901   
2  1901    Medicine  The Nobel Prize in Physiology or Medicine 1901   
3  1901       Peace                      The Nobel Peace Prize 1901   
4  1901       Peace                      The Nobel Peace Prize 1901   
5  1901     Physics                 The Nobel Prize in Physics 1901   
6  1902   Chemistry               The Nobel Prize in Chemistry 1902   
7  1902  Literature              The Nobel Prize in Literature 1902   
8  1902    Medicine  The Nobel Prize in Physiology or Medicine 1902   
9  1902       Peace                      The Nobel Peace Prize 1902   

                                          motivation prize_share  laureate_id  \
0  "in recognition of the extraordinary services ...         1/1          160   
1  "in special recognition of his poetic composit...         1/1          569   
2  "for his work on serum therapy, especially its...         1/1          293   
3                                                NaN         1/2          462   
4                                                NaN         1/2          463   
5  "in recognition of the extraordinary services ...         1/1            1   
6  "in recognition of the extraordinary services ...         1/1          161   
7  "the greatest living master of the art of hist...         1/1          571   
8  "for his work on malaria, by which he has show...         1/1          294   
9                                                NaN         1/2          464   

  laureate_type                           full_name  birth_date  \
0    Individual        Jacobus Henricus van 't Hoff  1852-08-30   
1    Individual                     Sully Prudhomme  1839-03-16   
2    Individual              Emil Adolf von Behring  1854-03-15   
3    Individual                   Jean Henry Dunant  1828-05-08   
4    Individual                      Frédéric Passy  1822-05-20   
5    Individual              Wilhelm Conrad Röntgen  1845-03-27   
6    Individual                Hermann Emil Fischer  1852-10-09   
7    Individual  Christian Matthias Theodor Mommsen  1817-11-30   
8    Individual                         Ronald Ross  1857-05-13   
9    Individual                       Élie Ducommun  1833-02-19   

           birth_city        birth_country   sex   organization_name  \
0           Rotterdam          Netherlands  Male   Berlin University   
1               Paris               France  Male                 NaN   
2   Hansdorf (Lawice)     Prussia (Poland)  Male  Marburg University   
3              Geneva          Switzerland  Male                 NaN   
4               Paris               France  Male                 NaN   
5  Lennep (Remscheid)    Prussia (Germany)  Male   Munich University   
6          Euskirchen    Prussia (Germany)  Male   Berlin University   
7             Garding  Schleswig (Germany)  Male                 NaN   
8              Almora                India  Male  University College   
9              Geneva          Switzerland  Male                 NaN   

  organization_city organization_country  death_date      death_city  \
0            Berlin              Germany  1911-03-01          Berlin   
1               NaN                  NaN  1907-09-07        Châtenay   
2           Marburg              Germany  1917-03-31         Marburg   
3               NaN                  NaN  1910-10-30          Heiden   
4               NaN                  NaN  1912-06-12           Paris   
5            Munich              Germany  1923-02-10          Munich   
6            Berlin              Germany  1919-07-15          Berlin   
7               NaN                  NaN  1903-11-01  Charlottenburg   
8         Liverpool       United Kingdom  1932-09-16    Putney Heath   
9               NaN                  NaN  1906-12-07            Bern   

    d

In [49]:
#tablo hakkında bilgi sahibi olabilmek için yapılan işlem. Boş değerler, türler, sütun isimleri, veri satır sayısı

nobel.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 911 entries, 0 to 910
Data columns (total 18 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   year                  911 non-null    int64 
 1   category              911 non-null    object
 2   prize                 911 non-null    object
 3   motivation            823 non-null    object
 4   prize_share           911 non-null    object
 5   laureate_id           911 non-null    int64 
 6   laureate_type         911 non-null    object
 7   full_name             911 non-null    object
 8   birth_date            883 non-null    object
 9   birth_city            883 non-null    object
 10  birth_country         885 non-null    object
 11  sex                   885 non-null    object
 12  organization_name     665 non-null    object
 13  organization_city     667 non-null    object
 14  organization_country  667 non-null    object
 15  death_date            593 non-null    ob

In [50]:
#sütun bazlı boş veri sayısı listesi

nobel.isnull().sum()

year                      0
category                  0
prize                     0
motivation               88
prize_share               0
laureate_id               0
laureate_type             0
full_name                 0
birth_date               28
birth_city               28
birth_country            26
sex                      26
organization_name       246
organization_city       244
organization_country    244
death_date              318
death_city              335
death_country           329
dtype: int64

In [51]:
#Analizi yaparken işimize yaramayacak sütunların bulunması

nobel = nobel.drop (["motivation","death_country","death_city","death_date","organization_country","organization_city","birth_city","laureate_id","prize_share","organization_name","laureate_type"], axis = 1)
nobel.head(10)

year    category                                           prize  \
0  1901   Chemistry               The Nobel Prize in Chemistry 1901   
1  1901  Literature              The Nobel Prize in Literature 1901   
2  1901    Medicine  The Nobel Prize in Physiology or Medicine 1901   
3  1901       Peace                      The Nobel Peace Prize 1901   
4  1901       Peace                      The Nobel Peace Prize 1901   
5  1901     Physics                 The Nobel Prize in Physics 1901   
6  1902   Chemistry               The Nobel Prize in Chemistry 1902   
7  1902  Literature              The Nobel Prize in Literature 1902   
8  1902    Medicine  The Nobel Prize in Physiology or Medicine 1902   
9  1902       Peace                      The Nobel Peace Prize 1902   

                            full_name  birth_date        birth_country   sex  
0        Jacobus Henricus van 't Hoff  1852-08-30          Netherlands  Male  
1                     Sully Prudhomme  1839-03-16               France  Male  
2              Emil Adolf von Behring  1854-03-15     Prussia (Poland)  Male  
3                   Jean Henry Dunant  1828-05-08          Switzerland  Male  
4                      Frédéric Passy  1822-05-20               France  Male  
5              Wilhelm Conrad Röntgen  1845-03-27    Prussia (Germany)  Male  
6                Hermann Emil Fischer  1852-10-09    Prussia (Germany)  Male  
7  Christian Matthias Theodor Mommsen  1817-11-30  Schleswig (Germany)  Male  
8                         Ronald Ross  1857-05-13                India  Male  
9                       Élie Ducommun  1833-02-19          Switzerland  Male

In [52]:
#kalan sütunların hepsinin türleri

nobel.dtypes

year              int64
category         object
prize            object
full_name        object
birth_date       object
birth_country    object
sex              object
dtype: object

In [53]:
#ve bu kalan sütunlarda boş yerlerin sütun bazın toplam listesi

nobel.isnull().sum()

year              0
category          0
prize             0
full_name         0
birth_date       28
birth_country    26
sex              26
dtype: int64

In [54]:
#boş sütunların hepsinin silinmesi

nobel.dropna(inplace=True)

In [55]:
nobel.isnull().sum()

year             0
category         0
prize            0
full_name        0
birth_date       0
birth_country    0
sex              0
dtype: int64

## TEMİZLEME VE HAZIRLAMA AŞAMASINDA YAPILAN ADIMLAR

- Google drive ile bağlantı kuruldu.
- Kullanılacak kütüphaneler yüklendi.
- Veri dosyası yüklendi.
- Veri hakkında tür, satır bilgisi, sütun bilgisi, boşluk bilgileri (NaN değerler) ve sütun tipleri hakkında bilgi alındı.
- Boş değerlerin sütunlarda ki toplam sayısına bakıldı.
- Analizi yaparken kullanılmayacak sütunların tamamı silindi.
- Kalan sütunların veri türlerine bakıldı.
- Tekrar toplamda kaç satırda ve hangi sütunlarda NaN değerler olduğuna bakıldı.
- Boş değerlerin olduğu satırların tamamı analiz yaparken yüksek sayıda olmadığı için ve sorun çıkarmayacağı tahmin edilerek silindi.
- Veri analizler için hazır.

## PROJE SORULARI VE CEVAPLARI

## 1. SORU - Nobel Ödüllerini en çok kazanan ilk on ülkeyi bulunuz.

In [56]:
#doğum üllkeleri bazında group by yapıp toplam sayısını alıyoruz
#arkasından en büyükten küçüğe sıralayıp sütun ismini değiştiriyorum ve sıralama numaralarını tablodan tekrar yazılıyor

top10 = nobel.groupby('birth_country').size().reset_index(name='Toplam').nlargest(10, 'Toplam').reset_index(drop=True)
top10

birth_country  Toplam
0  United States of America     258
1            United Kingdom      85
2                   Germany      61
3                    France      51
4                    Sweden      29
5                     Japan      24
6                    Canada      18
7               Netherlands      18
8                     Italy      17
9                    Russia      17

### CEVAP : Bu veriye göre ele alınan 115 yıllık veri üzerinde ABD, Birleşik Krallık, Almanya ve Fransa toplamda en çok Nobel ödülü almış ülkeler olmuş.

### ABD 6 farklı alanda toplamda 258 Nobel ödülü bulunuyor. ABD sırasıyla gelen diğer 6 tane ülkenin toplamına eşit geliyor.

###   Son 100 yılda yaşanan olaylarında etkisiyle ABD 'nin bir çok konuda gelişimi bu skora doğrudan etkisi olduğu net bir şekilde gözüküyor.



# 2. SORU - Nobel Ödüllerini kazanan ilk erkekleri listeleyiniz.


In [57]:
#male üzerined tabloun tamamını filtreleyip en küçükten büyüğe ilk 10 satırı yazdırıyor. Çıkan tabloya da yeni index numaraları getiriliyor.

nobel_male_firsttop10 = nobel[nobel['sex'] == 'Male'].nsmallest(10, 'year').reset_index(drop=True)
nobel_male_firsttop10

year    category                                           prize  \
0  1901   Chemistry               The Nobel Prize in Chemistry 1901   
1  1901  Literature              The Nobel Prize in Literature 1901   
2  1901    Medicine  The Nobel Prize in Physiology or Medicine 1901   
3  1901       Peace                      The Nobel Peace Prize 1901   
4  1901       Peace                      The Nobel Peace Prize 1901   
5  1901     Physics                 The Nobel Prize in Physics 1901   
6  1902   Chemistry               The Nobel Prize in Chemistry 1902   
7  1902  Literature              The Nobel Prize in Literature 1902   
8  1902    Medicine  The Nobel Prize in Physiology or Medicine 1902   
9  1902       Peace                      The Nobel Peace Prize 1902   

                            full_name  birth_date        birth_country   sex  
0        Jacobus Henricus van 't Hoff  1852-08-30          Netherlands  Male  
1                     Sully Prudhomme  1839-03-16               France  Male  
2              Emil Adolf von Behring  1854-03-15     Prussia (Poland)  Male  
3                   Jean Henry Dunant  1828-05-08          Switzerland  Male  
4                      Frédéric Passy  1822-05-20               France  Male  
5              Wilhelm Conrad Röntgen  1845-03-27    Prussia (Germany)  Male  
6                Hermann Emil Fischer  1852-10-09    Prussia (Germany)  Male  
7  Christian Matthias Theodor Mommsen  1817-11-30  Schleswig (Germany)  Male  
8                         Ronald Ross  1857-05-13                India  Male  
9                       Élie Ducommun  1833-02-19          Switzerland  Male

### CEVAP : ilk kazanan erkekler listesinde ilk dikkat çeken Edebiyat kategorisinin olmaması. Ayrıca 1. yılında Barış ödülünün de 2 kişiye verilmesi oldu.

### Bunun yanın da 1 . yılında ödül sadece avrupa ülkelerine verilmiş.


# 3. SORU - Nobel Ödüllerini kazanan ilk kadınları listeleyiniz.

In [58]:
# 3. soru ve 2. soru aynı kod mantığı var.

nobel_female_firsttop10 = nobel[nobel['sex'] == 'Female'].nsmallest(10, 'year').reset_index(drop=True)
nobel_female_firsttop10

year    category                               prize  \
0  1903     Physics     The Nobel Prize in Physics 1903   
1  1905       Peace          The Nobel Peace Prize 1905   
2  1909  Literature  The Nobel Prize in Literature 1909   
3  1911   Chemistry   The Nobel Prize in Chemistry 1911   
4  1926  Literature  The Nobel Prize in Literature 1926   
5  1928  Literature  The Nobel Prize in Literature 1928   
6  1931       Peace          The Nobel Peace Prize 1931   
7  1935   Chemistry   The Nobel Prize in Chemistry 1935   
8  1938  Literature  The Nobel Prize in Literature 1938   
9  1945  Literature  The Nobel Prize in Literature 1945   

                                           full_name  birth_date  \
0                        Marie Curie, née Sklodowska  1867-11-07   
1  Baroness Bertha Sophie Felicita von Suttner, n...  1843-06-09   
2                      Selma Ottilia Lovisa Lagerlöf  1858-11-20   
3                        Marie Curie, née Sklodowska  1867-11-07   
4                                     Grazia Deledda  1871-09-27   
5                                      Sigrid Undset  1882-05-20   
6                                        Jane Addams  1860-09-06   
7                                 Irène Joliot-Curie  1897-09-12   
8                                         Pearl Buck  1892-06-26   
9                                   Gabriela Mistral  1889-04-07   

                      birth_country     sex  
0           Russian Empire (Poland)  Female  
1  Austrian Empire (Czech Republic)  Female  
2                            Sweden  Female  
3           Russian Empire (Poland)  Female  
4                             Italy  Female  
5                           Denmark  Female  
6          United States of America  Female  
7                            France  Female  
8          United States of America  Female  
9                             Chile  Female

### CEVAP: İlk 10 kazanan kadın ödül sahiplerinin listesine baktığımızda ilk görünen ilk ödülün 1903 ve 10. ödülün 1945 yılları arasında verildiği ve 42 yıl içinde kadın kazananların bir hayli az aldığı görülüyor.

### Kadınların ilk 40-45 yılda bu kadar az almasını sebeplerinden bir tanesi 1.ve 2. Dünya savaşları olabilir.

### Ayrıca, yüksek ve genel eğitimin (okuma ve yazma) de kadın ve erkek eşitliğinde verilmemesi olabilir.


# 4. SORU - Nobel ödülünü en çok kazanan ülkenin hangi yıldan itibaren hakimiyet sağladığını görselleştirip bu hakimiyette rol oynayan şeyler nelerdir? İçgörülerinizi paylaşır mısınız?

In [59]:
# 2. 3. soruların mantığıyla aynı olarak USA üzerinde filtreleme yapılmıştır. Tablo kendini küçükten büyüğe yıl bazlı sıralamıştır.
nobel_winner_usa = nobel[nobel['birth_country'] == 'United States of America'].reset_index(drop=True)
nobel_winner_usa

year    category                                           prize  \
0    1906       Peace                      The Nobel Peace Prize 1906   
1    1912       Peace                      The Nobel Peace Prize 1912   
2    1914   Chemistry               The Nobel Prize in Chemistry 1914   
3    1919       Peace                      The Nobel Peace Prize 1919   
4    1923     Physics                 The Nobel Prize in Physics 1923   
..    ...         ...                                             ...   
253  2014   Chemistry               The Nobel Prize in Chemistry 2014   
254  2014   Chemistry               The Nobel Prize in Chemistry 2014   
255  2014    Medicine  The Nobel Prize in Physiology or Medicine 2014   
256  2015   Chemistry               The Nobel Prize in Chemistry 2015   
257  2016  Literature              The Nobel Prize in Literature 2016   

                     full_name  birth_date             birth_country   sex  
0           Theodore Roosevelt  1858-10-27  United States of America  Male  
1                   Elihu Root  1845-02-15  United States of America  Male  
2    Theodore William Richards  1868-01-31  United States of America  Male  
3        Thomas Woodrow Wilson  1856-12-28  United States of America  Male  
4      Robert Andrews Millikan  1868-03-22  United States of America  Male  
..                         ...         ...                       ...   ...  
253                Eric Betzig  1960-01-13  United States of America  Male  
254         William E. Moerner  1953-06-24  United States of America  Male  
255               John O'Keefe  1939-11-18  United States of America  Male  
256               Paul Modrich  1946-06-13  United States of America  Male  
257                  Bob Dylan  1941-05-24  United States of America  Male  

[258 rows x 7 columns]

In [60]:
#listenin kategori bazında kazandığı kategorileri getirecek.

nobel_winner_usa_grouped = nobel_winner_usa.groupby(['category']).size().reset_index(name='count')

nobel_winner_usa_grouped

category  count
0   Chemistry     52
1   Economics     43
2  Literature      9
3    Medicine     70
4       Peace     19
5     Physics     65

In [61]:
#histogram grafiği ile yıllar üzerinde grup yapıldı ve her 10 yılda toplam kaçar nobel kazanılmış gröseli çıkarıldı.

fig1 = px.histogram(nobel_winner_usa, x='year',title="ABD'NİN 10'ar YILLIK NOBEL KAZANANLARI (1901-2016)")


fig1.update_layout(
    title=dict(text="ABD'NİN 10'ar YILLIK NOBEL KAZANANLARI (1901-2016)", font=dict(size=24)),
)

fig1.show()

## CEVAP - ABD'nin kazandığı Nobel ödüllerine bakıldığında 1930 dan itibaren hızlı bir şekilde katlanarak arttığı görülüyor.

## En önemli sebeplerinden bir tanesi 1920 lerde yaşadığı ekonomik ve sosyal problemlerin arkasından 1930lar da avrupa da ki savaşlar yüzünden göç alması, sanayi de gelişmelerin olması ve diğer ülkeler savaşırken gelişimin merkezinde olması. Diğer olumsuz faaliyetlerden de uzak kalması denilebilir.  

## 1940'larda Avrupa'da ki dünya savaşının getridiği büyük beyin göçü, insanların daha güvenli kıtalara göç etmesinin çok büyük etkeni olduğu söylenebilir. 1940 larda ki ufak düşüşün açıklaması dünya nüfusuna olduğu kadar ABD'ye de etkisinin çok yüksek olduğu Dünya savaşı olabilir.

## 1950 yıllarında 1940 larda ülkesine aldığı büyük beyin göçünün meyvelerini toplamaya devam etmesidir. Bu dönemde uzay biliminde de yaptığı büyük gelişmeler bulunyor. 1955-1975 arasında devam eden veitnam savaşının getirdiği olumsuz etkilenmeler 1980 yılında düşüş olarak görülebilir.

## Bu katlanarak büyüyen Nobel ödülü sayılarında ki düşüş 2010 yılına kadar sürmüştür.

## Avrupa'nın son 10-15 yılda ki tekrar yeni merkez olması ve daha çok göçün ve eğitim seviyesinin bu topraklarda artması büyük etken yaratmıştır.

# 5. SORU - Nobel Ödülü kazananların cinsiyetlerini, yaşlarını, ödül kategorisi ve yılları kullanarak görselleştiriniz.(Her bir ödül kategorisi için ayrı grafik gösteriniz) Çıkan sonuçlara göre görseli yorumlayınız.

In [62]:
#öncelikle nobel tablosunun kopyası oluşturuldu. birthdate sütunundan yıl sütununu çıkarmamız gerekiyor çünkü kazanma yaşının sütununu yapacağız ve bu sayılara göre de bir iç göreü çıkaracağız.
#birthdate veri tipi stringten birthdate türüne çevrildi ve çıkarma işlemi yapıldı.

nobel1 = nobel.copy()
nobel1['birth_date'] = pd.to_datetime(nobel1['birth_date'], errors='coerce')
nobel1['winner_age'] = nobel1['year'] - nobel1['birth_date'].dt.year

nobel1

year    category                                           prize  \
0    1901   Chemistry               The Nobel Prize in Chemistry 1901   
1    1901  Literature              The Nobel Prize in Literature 1901   
2    1901    Medicine  The Nobel Prize in Physiology or Medicine 1901   
3    1901       Peace                      The Nobel Peace Prize 1901   
4    1901       Peace                      The Nobel Peace Prize 1901   
..    ...         ...                                             ...   
906  2016    Medicine  The Nobel Prize in Physiology or Medicine 2016   
907  2016       Peace                      The Nobel Peace Prize 2016   
908  2016     Physics                 The Nobel Prize in Physics 2016   
909  2016     Physics                 The Nobel Prize in Physics 2016   
910  2016     Physics                 The Nobel Prize in Physics 2016   

                        full_name birth_date     birth_country   sex  \
0    Jacobus Henricus van 't Hoff 1852-08-30       Netherlands  Male   
1                 Sully Prudhomme 1839-03-16            France  Male   
2          Emil Adolf von Behring 1854-03-15  Prussia (Poland)  Male   
3               Jean Henry Dunant 1828-05-08       Switzerland  Male   
4                  Frédéric Passy 1822-05-20            France  Male   
..                            ...        ...               ...   ...   
906              Yoshinori Ohsumi 1945-02-09             Japan  Male   
907            Juan Manuel Santos 1951-08-10          Colombia  Male   
908             David J. Thouless 1934-09-21    United Kingdom  Male   
909          F. Duncan M. Haldane 1951-09-14    United Kingdom  Male   
910         J. Michael Kosterlitz 1943-06-22    United Kingdom  Male   

     winner_age  
0            49  
1            62  
2            47  
3            73  
4            79  
..          ...  
906          71  
907          65  
908          82  
909          65  
910          73  

[883 rows x 8 columns]

In [63]:
#yukarıda ki tabloda ki veri lerin x eksenine kazanım yılı ve y eksenine kazanma yaşını koyarak scatter plot grafiği yapıldı.
#Erkek ve Kadın kazananlarda iki ayrı satırda gösterildi.
#kategoriler sütun halinde gösterildi.
#yazım sıralaması da belirtildi.
#ve başlık eklendi.

fig2 = px.scatter(nobel1, x="year", y='winner_age', color='sex', height=700, width=3000,
                  facet_row='sex', facet_col="category",
                  category_orders={"sex": ["Male", "Female"],
                                   "category": ["Chemistry", "Literature", "Medicine", "Peace", "Physics", "Economics"]},
                  labels={"sex": "Cinsiyet", "category": "Kategori", "year": "Yıl", "winner_age": "Kazanan Yaş"},
                  title="Nobel Ödül Kazananların Yaşları ve Kategorilerine Göre Dağılımı")

fig2.update_layout(
    title=dict(text="Nobel Ödül Kazananların Yaşları ve Kategorilerine Göre Dağılımı", font=dict(size=24)),
)

fig2.show()

## CEVAP - Kategorilerin hepsini teker teker değerlendirmesi yapılırsa;

## Kimya kategorisinde oran olarak Erkek kazananların sayısı Kadınlardan 10 veya 11 kat daha fazla olduğu dağılım grafiiğinde gözüküyor. 115 yıl içinde kadın kazanan sayısı 4 adet olduğu görülüyor. Bilim kadınlarının son 100 yıl içinde bu konuda bir ödülü olmadığı görülüyor ve bu kategoride kazananların yaşları 40 ile 80 yaşları arasında olduğu da gözüküyor.

## Edebiyat alanında kazanılan ödüllerde kadınların 90'lar ve 2000 lerde bir çok ödül aldığı görülüyor. Kadın yazarların Edebiyat alanında kazanılan ödüllerin de son 15 yılda 2 katına çıktığı görülüyor. Bu gösterge kadın yazarlarının bir çok ülkede artması olabilir.  

## Tıp alanında kazanılan ödüllerde erkekler tekrar yüksek üstünlüğe sahip. Fakat, son 30 yılda Kadın bilim insanlarında görülen bir artış var. Kadınların yüksek ve genel eğitimde dünya çapında sayı olarak yükselmesi bunun başlıca sebeplerinden biri olabilir.

## Barış alanında kadın kazananlar arasında yaşça küçük son yıllarda bir kaç kazanan olmuş. Son 30 yılda yaşanan olayların yaşça küçük bireyler tarafından insanlık tarihinde iz bırakacak hareketler yapması ödüllendirilmiş ve tarihe kadın barış nobel ödülü kazananları olarak geçmiş.

## Fizik alanında erkeklerin Nobel ödüllerinin 115 yıllık tarihi boyunca bir çok defa kazanmış. Kadın bilim insanları sadece 2 defa kazanmış. Sayıca az bile olsa kadın düşünürlerin dünyanın bir çok alanında olması gelişimin ilk adımlarıdır. Kadınların bu dalda ki kazanma rakamlarının düşük olması genelde fizik bölümlerinde düşünülen kadar çok kadın öğrencinin olmaması olabilir.   

## Genel değerlendirme yapılırsa; Kadın bilim insanları ve düşüürlerin son 30 yılda kazanım sayılarının yükseldiği görülüyor. Bundan sonra ki 30-50 yıl arasında da artış olacağını tahmin edebiliriz. Savaşlar, ekonomik sıkıtılar, göç ve eğitim bu ödüllerin kazanan ülkelerinin sayılarını doğrudan etkileyen faktörlerdendir.


# 6. SORU 1938-1945 yılı arasında Nobel Ödülü kazananların kategorilerini ve ülkelerini görselleştirip yorumlayınız.

In [64]:
nobel.dtypes

year              int64
category         object
prize            object
full_name        object
birth_date       object
birth_country    object
sex              object
dtype: object

In [65]:
#en başta çıkardığımız tablodan yine yeni bir tablo yaratarak sütun silindi. axis = 1 yazılması gerekiyor çünkü yukarıdan aşağı silinmesi gerektiğinin komutu yazılmalıdır.

nobel2 = nobel.drop(["prize","birth_date","sex","full_name"], axis = 1)

nobel2

year    category     birth_country
0    1901   Chemistry       Netherlands
1    1901  Literature            France
2    1901    Medicine  Prussia (Poland)
3    1901       Peace       Switzerland
4    1901       Peace            France
..    ...         ...               ...
906  2016    Medicine             Japan
907  2016       Peace          Colombia
908  2016     Physics    United Kingdom
909  2016     Physics    United Kingdom
910  2016     Physics    United Kingdom

[883 rows x 3 columns]

In [66]:
nobel2.dtypes

year              int64
category         object
birth_country    object
dtype: object

In [67]:
#filtreleme işlemi yapıldı ve belirtilen zamanlar arsında ki veriler alındı. index te tekrar yazıldı.

nobel2_filtered_38_45 = nobel2[(nobel2['year'] >= 1938) & (nobel2['year'] <= 1945)].reset_index(drop=True)

nobel2_filtered_38_45

year    category              birth_country
0   1938   Chemistry  Austria-Hungary (Austria)
1   1938  Literature   United States of America
2   1938    Medicine                    Belgium
3   1938     Physics                      Italy
4   1939   Chemistry                    Germany
5   1939   Chemistry  Austria-Hungary (Croatia)
6   1939  Literature   Russian Empire (Finland)
7   1939    Medicine           Germany (Poland)
8   1939     Physics   United States of America
9   1943   Chemistry  Austria-Hungary (Hungary)
10  1943    Medicine                    Denmark
11  1943    Medicine   United States of America
12  1943     Physics           Germany (Poland)
13  1944   Chemistry                    Germany
14  1944  Literature                    Denmark
15  1944    Medicine   United States of America
16  1944    Medicine   United States of America
17  1944     Physics   Austria-Hungary (Poland)
18  1945   Chemistry   Russian Empire (Finland)
19  1945  Literature                      Chile
20  1945    Medicine                   Scotland
21  1945    Medicine                    Germany
22  1945    Medicine                  Australia
23  1945       Peace   United States of America
24  1945     Physics                    Austria

In [68]:
nobel2_filtered_38_45.shape

(25, 3)

In [69]:
#bar grafiği kullanıldı. ülkeler x eksenine getirildi ve kategorilerde toplanarak ülkeler üzerine  farklı kategori renkleri ile sıralandı.

fig3 = px.bar(nobel2_filtered_38_45, x="birth_country", color="category", title="1938-1945 ÜLKELER BAZLI KAZANAN KATEGORİLER")
fig3.update_layout(
    title=dict(text="1938-1945 ÜLKELER BAZLI KAZANAN KATEGORİLER", font=dict(size=24)),
)
fig3.show()

## CEVAP - 2. Dünya savaşının başlangıç ve bitiş tarihleri arasında verilen Nobel ödüllerinde Avrupa ülkelerinin toplamda büyük bir üstünlüğü var. ABD 'nin toplamda aldığı ödüllerde Avusturya - Macaristan krallığının vatandaşları toplamında eşitlik görülüyor.

## ABD 'nin Tıp kategorisinde bu 7 senede aldıkları 3 farklı ödül ile diğer ülkelere tek başına fark atmış görülüyor.

## Barış, fizik, Edebiyat ve Tıp kategorilerinde ödül alan ABD bu dönemin yükselişte olan ülkesi olarak görülüyor. Bilim insanlarının göç akınına uğrayan ve 2. dünya savaşından kaçanların çoğunlukta olduğu gözlenebilir.


# 7. SORU - 1947-1991 yılları arasında Nobel Ödülü kazananların kategorilerini ve ülkelerini görselleştirip yorumlayınız.(Her kategori için ayrı bir grafik olması istenmektedir)



In [70]:
#filtreleme işlemi yapıldı.

nobel2_filtered_47_91 = nobel2[(nobel2['year'] >= 1947) & (nobel2['year'] <= 1991)].reset_index(drop=True)

nobel2_filtered_47_91

year    category                     birth_country
0    1947   Chemistry                    United Kingdom
1    1947  Literature                            France
2    1947    Medicine  Austria-Hungary (Czech Republic)
3    1947    Medicine  Austria-Hungary (Czech Republic)
4    1947    Medicine                         Argentina
..    ...         ...                               ...
372  1991  Literature                      South Africa
373  1991    Medicine                           Germany
374  1991    Medicine                           Germany
375  1991       Peace                   Burma (Myanmar)
376  1991     Physics                            France

[377 rows x 3 columns]

In [71]:
#en çok kazanan ülke verisinde bu tarih aralığında yüksekten küçüğe toplandı ve tablo yazdırıldı.

nobel2_filtered_47_91_grouped = nobel2_filtered_47_91.groupby('birth_country').size().reset_index(name='Toplam').nlargest(10, 'Toplam')
nobel2_filtered_47_91_grouped

birth_country  Toplam
73  United States of America     116
72            United Kingdom      39
24                   Germany      30
21                    France      18
67                    Sweden      15
55                    Russia      10
35                     Italy       9
13                    Canada       8
68               Switzerland       8
36                     Japan       7

In [72]:
# 6 kategoriyi de farklı gruplandırark grafiklerde karşılaştırmak amacıyla en iyi görseli treemap yapacağı düşünüldü.
# Kategorilerin hepsi farklı değerlendirildi ve kendi içlerinde toplamları gösterildi.

nobel2_filtered_47_91_grouped = nobel2_filtered_47_91.groupby(['category', 'birth_country']).size().reset_index(name='count')


fig5 = px.treemap(nobel2_filtered_47_91_grouped, path=['category', 'birth_country'], title="ÜLKE BAZINDA KAZANAN KATEGORİLER 1947-1991",
                  values='count',
                  color='count',
                  hover_data=['birth_country'],
                  color_continuous_scale='Mrybm')

fig5.update_layout(margin=dict(t=65, l=45, r=25, b=45))
fig5.update_layout(title_x=0.5)
fig5.update_layout(
    title=dict(text="ÜLKE BAZINDA KAZANAN KATEGORİLER 1947-1991", font=dict(size=24)),
)
fig5.show()


## CEVAP - Öncelikle tarih aralığı 1947 ile 1991 olarak incelenilmesi istenmiş. Bu aralığın 2. Dünya savaşından sonra olduğu için ülkelerin kazanma miktarları gelişmin ve toplumun bir çok alanda yeniden oluşumunun ilerlemesinde etkisi olduğu için bu grafik ülkeler bazında yorumlanabilir.

## ilk bakışta kategorilerin isimleri dışında göze çarpan ülke isimleri ABD, Almanya, Birleşik Krallık ve Fransa görülüyor. Tıp, Fizik, Kimya ve Ekonomi alanlarlarında toplam sayılarda ilk 5 ülke arasında görülüyor.

## Bu konularda başlıca bu ülkelerin yukarı sıralarda olması dünya etrafından özellikle ABD 'nin içine aldığı göç miktarının 2. dünya savaşı sonrasında tekrar yapılanmanın verdiği düzenli ve önü açık eğitimin direk etki yaratması olabilir. Yüksek öğretim de ki katkıları ve sosyo kültürel anlamda ki katkısı toplamun gelişimde ve ilerlemesinde teknoloji, bilim ve Ekonomi alanlarında büyük gelişmeler olmuştur.

## Edebiyat ve Barış alanlarında ilk 10 da diğer bilim dallarından farklı bir görüntü bulunuyor. Edebiyatta alınan ödüllerde UK, ABD, Fransa, Rusya İspanya ve isveç var. Edebiyat alanında alınan bu ödüllerin neredeyse tamamının avrupa ülkelerinden ve ABD'nin de bu yıllar içinde aldığı büyük göç yüzünden en sıralarda olduğu görülebiliyor.

## Barış ödülünün de bir çok farklı ülkeden kazananı olması Nobel ödüllerininde uluslararası politik ilişkilerin de büyük etkisi olduğunu söyleyebilinir. Özellikle bu yıllar içinde Avrupa'da ki birlik berberliğin kurulmasının şart olunduğu zamanlarda.







# 8. SORU - Kimya, Edebiyat, Barış, Fizik ve Tıp kategorilerindeki 2000 sonrasındaki kişilerin ülkelerini, yaşlarını görselleştirin.(Her bir Kategori için ayrı görselleştirme yapılması istenmektedir) Veriyi yorumlayınız.

In [73]:
nobel1

year    category                                           prize  \
0    1901   Chemistry               The Nobel Prize in Chemistry 1901   
1    1901  Literature              The Nobel Prize in Literature 1901   
2    1901    Medicine  The Nobel Prize in Physiology or Medicine 1901   
3    1901       Peace                      The Nobel Peace Prize 1901   
4    1901       Peace                      The Nobel Peace Prize 1901   
..    ...         ...                                             ...   
906  2016    Medicine  The Nobel Prize in Physiology or Medicine 2016   
907  2016       Peace                      The Nobel Peace Prize 2016   
908  2016     Physics                 The Nobel Prize in Physics 2016   
909  2016     Physics                 The Nobel Prize in Physics 2016   
910  2016     Physics                 The Nobel Prize in Physics 2016   

                        full_name birth_date     birth_country   sex  \
0    Jacobus Henricus van 't Hoff 1852-08-30       Netherlands  Male   
1                 Sully Prudhomme 1839-03-16            France  Male   
2          Emil Adolf von Behring 1854-03-15  Prussia (Poland)  Male   
3               Jean Henry Dunant 1828-05-08       Switzerland  Male   
4                  Frédéric Passy 1822-05-20            France  Male   
..                            ...        ...               ...   ...   
906              Yoshinori Ohsumi 1945-02-09             Japan  Male   
907            Juan Manuel Santos 1951-08-10          Colombia  Male   
908             David J. Thouless 1934-09-21    United Kingdom  Male   
909          F. Duncan M. Haldane 1951-09-14    United Kingdom  Male   
910         J. Michael Kosterlitz 1943-06-22    United Kingdom  Male   

     winner_age  
0            49  
1            62  
2            47  
3            73  
4            79  
..          ...  
906          71  
907          65  
908          82  
909          65  
910          73  

[883 rows x 8 columns]

In [74]:
#gereksiz kolonlar tekrar silindi.

nobel3 = nobel1.drop(["prize","full_name","birth_date","sex"], axis = 1)

nobel3

year    category     birth_country  winner_age
0    1901   Chemistry       Netherlands          49
1    1901  Literature            France          62
2    1901    Medicine  Prussia (Poland)          47
3    1901       Peace       Switzerland          73
4    1901       Peace            France          79
..    ...         ...               ...         ...
906  2016    Medicine             Japan          71
907  2016       Peace          Colombia          65
908  2016     Physics    United Kingdom          82
909  2016     Physics    United Kingdom          65
910  2016     Physics    United Kingdom          73

[883 rows x 4 columns]

In [75]:
#5 kategori üzerinde filtreleme işlemi yapıldı.

nobel3_filtered = nobel3[(nobel3['year'] >= 2000) & ((nobel3['category'] == "Chemistry") |
                                                      (nobel3['category'] == "Literature") |
                                                        (nobel3['category'] == "Peace") |
                                                         (nobel3['category'] == "Medicine") |
                                                          (nobel3['category'] == "Physics"))].reset_index(drop=True)

nobel3_filtered

year    category             birth_country  winner_age
0    2000   Chemistry  United States of America          64
1    2000   Chemistry               New Zealand          73
2    2000   Chemistry                     Japan          64
3    2000  Literature                     China          60
4    2000    Medicine                    Sweden          77
..    ...         ...                       ...         ...
157  2016    Medicine                     Japan          71
158  2016       Peace                  Colombia          65
159  2016     Physics            United Kingdom          82
160  2016     Physics            United Kingdom          65
161  2016     Physics            United Kingdom          73

[162 rows x 4 columns]

In [76]:
#yaşlar üzerinde bir iç görü çıkarmak için tablo açıklaması çıkarıldı.

nobel3_filtered.describe()

year  winner_age
count   162.000000  162.000000
mean   2007.802469   65.722222
std       5.022730   12.282012
min    2000.000000   17.000000
25%    2003.000000   58.000000
50%    2008.000000   67.000000
75%    2012.000000   75.000000
max    2016.000000   88.000000

In [77]:
#ekstra grouplama yapıp önce kategori sonra doğum ülkesi ve yaşı üzerinde gruplandırıldı. Fakat bu veri görselleştirilmedi.
nobel3_filtered_grouped = nobel3_filtered.groupby(['category', 'birth_country','winner_age']).size().reset_index(name='count')

nobel3_filtered_grouped

category                               birth_country  winner_age  count
0    Chemistry                                     Austria          83      1
1    Chemistry                                     Belgium          75      1
2    Chemistry       British Mandate of Palestine (Israel)          70      2
3    Chemistry       British Mandate of Palestine (Israel)          73      1
4    Chemistry  British Protectorate of Palestine (Israel)          57      1
..         ...                                         ...         ...    ...
148    Physics                    United States of America          77      1
149    Physics                    United States of America          79      1
150    Physics                    United States of America          80      1
151    Physics                    United States of America          88      1
152    Physics                      West Germany (Germany)          44      1

[153 rows x 4 columns]

In [78]:
nobel3_filtered_grouped = nobel3_filtered.groupby(['category', 'birth_country']).agg({'winner_age': 'mean'}).reset_index()

fig11 = px.bar(nobel3_filtered_grouped, x='birth_country', y='winner_age', color='category', height=1200, width=3000,
             facet_row='category',
             category_orders={"category": ["Chemistry", "Literature", "Peace", "Physics", "Medicine"]},
             labels={'winner_age': 'Yaş', 'birth_country': 'Doğum Ülkesi'},
             title='2000 Sonrasındaki Nobel Kazananlarının Ülkeleri ve Yaşları')
fig11.update_layout(
    title=dict(text='2000 Sonrasındaki Nobel Ödülü Kazananlarının Ülkeleri ve Yaşları', font=dict(size=24)),
)
fig11.show()


## CEVAP - ilk bakışta dikkat çeken özellik hemen hemen bütün kategorilerde kazananların yaşlarının 50 yaşın üzerinde görüldüğüdür.

## Bu tablonun ortalama yaşının 65 olarak gösterdiği tabloda, yaşı 20 yaşlarında olanlar sadece Barış ödülü kategorisinde bulunmaktadır. Kazanan ödül sahiplerinin de doğum ve yüksek öğretime kadar olan zamanı 2. dünya savaşı sonrası gerçekleşen büyük değişim içinde büyütülen ve eğitim gören düşünür ve öncüler olduğu görülüyor.

## Bütün ödül almış ülkelerde ilginç bir patern olarak görülecek bir durumda: bilim dalları olan kimya, fizik ve tıp alanlarında Dünya gayri safi milli hasılasının yüksek olan ve yatırım için kuvvetli olan ülkeleri (Avrupa ülkeleri, Kuzey Amerika, Uzak doğu) bulunuyor. Edebiyat ve Barış ödülleri bu kuvveti düşük olan veya hiç kuramayan ülkelerde bulunuyor (Ortadoğu, Afrika ve Güney Amerika). Bu durumun sebebi politik sebepler olduğunu düşünebiliriz.

## Bütün dallarda ödül almış sadece 2 ülke gözüküyor. Bunlar Çin ve ABD olarak gözüküyor. Ülkelerin değerlendirmelerini yaparken belirtildiği gibi ticaret ve üretim kanallarında yapılan değişim ve yapılanmaların başlangıcı olabilir.  



# ---------  GENEL DEĞERLENDİRME  ------------


## Nobel ödüllerinin her yıl farklı ülkelere gitmesinin ekonomik, sosyal, çevresel, politik, teknolojik ve refah seviyesi yüksek veya düşük olmasına direk bağlantılı olduğunu düşünmemenin mümkün olmadığı söylenebilir. Avrupa ülkelerinde yaşanan büyük değişimlerin getirdiği etkiler, ekonomik seviyede ticaret kanallarının artması, birleşmiş milletlerin kurulması ve birlik bilinciniin artması, üretimin bilgi ve yeni enerji buluşlarıyla hızla yükselmesi, bilginin belirli ülkelerde diğerlerine göre daha kolay ulaşılabilirliği, yatırım imkanları ve gelişim bilinci ülkeler arasında ki farkılılıkları bu ödüllerin sonuç analizinde de izlenebiliyor.